pkg/R/sparkR.R

```diff
@@ -52,6 +60,17 @@ sparkR.init <- function(
     return(get(".sparkRjsc", envir=.sparkREnv))
   }
 
+  sparkMem <- Sys.getenv("SPARK_MEM", "512m")
+  cp = .sparkREnv$assemblyJarPath
+  yarn_conf_dir <- Sys.getenv("YARN_CONF_DIR", "")
+  if (yarn_conf_dir != "") {
+    cp = paste(cp, yarn_conf_dir, sep=":")
+  }
+  launchBackend(classPath=cp, mainClass="edu.berkeley.cs.amplab.sparkr.SparkRBackend",
+                args="12345", javaOpts=paste("-Xmx", sparkMem, sep=""))
```

Comment:
could we add a configuration allowing the user to specify a port in the case of port conflict?